### PR TITLE
Move charts/Makefile targets to root Makefile and third_party 

### LINKED ISSUE #515: chore: move charts/Makefile targets to root Makefile and third_party
* cert-manager should move to third_party
* vault should move to third_party
* m4d should move to root
* run-integration-tests should be removed

If splitting install from wait actually helps to CI performance we should do the split as we had before

### PR DESCRIPTION
**Pull request by revit13**
This PR moves targets from charts/Makefile as follows:
1) cert-manager moves to third_party/ directory
2) vault and related targets move to third_party/ directory
3) m4d moves to root Makefile
4) run-integration-tests is removed

this pr also removes unused files from third_part/vault directory.

Closes #515

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,6 +9,15 @@ license: $(TOOLBIN)/license_finder
 docker-mirror-read:
 	$(TOOLS_DIR)/docker_mirror.sh $(TOOLS_DIR)/docker_mirror.conf
 
+.PHONY: m4d
```
Comment (roee88) [inline Makefile:12]: Consider renaming `m4d` to something else like deploy or something

File: Makefile [hunk 2]
```diff
@@ -9,6 +9,15 @@ license: $(TOOLBIN)/license_finder
...
+m4d:    export M4D_VALUES_FILE?=charts/m4d/values.yaml
+m4d:
+	kubectl create namespace m4d-system || true
```
Comment (roee88) [inline Makefile:15]: Consider using a variable for `m4d-system` at the top of this makefile

Comment (roee88) [inline Makefile:15]: Use `$(TOOLBIN)` for kubectl and helm. Also add them to the make target like
```
m4d: $(TOOLBIN)/kubectl $(TOOLBIN)/helm
```

File: third_party/cert-manager/Makefile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+ROOT_DIR:=../..
+include $(ROOT_DIR)/Makefile.env
+include $(ROOT_DIR)/hack/make-rules/tools.mk
+
+WITH_OPENSHIFT ?= false
+
+.PHONY: deploy
+deploy:
```
Comment (roee88) [inline third_party/cert-manager/Makefile:8]: Same comment on adding TOOLBIN dependencies to the target (also in vault)

File: Makefile [hunk 4]
```diff
@@ -18,17 +28,17 @@ test:
...
 run-integration-tests: export DOCKER_NAMESPACE?=m4d-system
-run-integration-tests: export VALUES_FILE=m4d/integration-tests.values.yaml
+run-integration-tests: export VALUES_FILE=charts/m4d/integration-tests.values.yaml
 run-integration-tests:
 	$(MAKE) kind
-	$(MAKE) -C charts vault
-	$(MAKE) -C charts wait-for-vault
-	$(MAKE) -C charts cert-manager
+	$(MAKE) -C third_party/vault deploy
+	$(MAKE) -C third_party/vault wait-for-vault
```
Comment (roee88) [inline Makefile:35]: Either merge these two so `vault deploy` will do the wait like what we do for cert-manager, or do the wait in `cluster-prepare-wait` like what we do for datashim

File: Makefile [hunk 5]
```diff
@@ -18,17 +28,17 @@ test:
...
+	$(MAKE) -C third_party/cert-manager deploy
 	$(MAKE) -C third_party/datashim deploy
 	$(MAKE) docker
```
Comment (roee88) [inline Makefile:38]: Consider removing `docker` target and using here `$(MAKE) docker-build docker-push` directly

File: Makefile [hunk 6]
```diff
@@ -18,17 +28,17 @@ test:
...
 	$(MAKE) -C test/services docker-build docker-push
 	$(MAKE) cluster-prepare-wait
-	$(MAKE) -C charts m4d
+	$(MAKE) deploy
 	$(MAKE) -C manager wait_for_manager
```
Comment (roee88) [inline Makefile:42]: I think this can be removed because deploy already waits

File: Makefile [hunk 7]
```diff
@@ -18,17 +28,17 @@ test:
...
 	$(MAKE) configure-vault
 	$(MAKE) helm
```
Comment (roee88) [inline Makefile:44]: Consider removing the `helm` target and use here `$(MAKE) -C modules helm` directly. 
(I don't think `helm` is descriptive enough but that's another issue)

File: Makefile [hunk 8]
```diff
@@ -18,17 +28,17 @@ test:
...
+run-integration-tests: export VALUES_FILE=charts/m4d/integration-tests.values.yaml
 run-integration-tests:
 	$(MAKE) kind
-	$(MAKE) -C charts vault
-	$(MAKE) -C charts wait-for-vault
-	$(MAKE) -C charts cert-manager
+	$(MAKE) -C third_party/vault deploy
+	$(MAKE) -C third_party/vault wait-for-vault
+	$(MAKE) -C third_party/cert-manager deploy
 	$(MAKE) -C third_party/datashim deploy
```
Comment (roee88) [inline Makefile:37]: Consider reusing `cluster-prepare` if it does the same

**Review by roee88 — COMMENTED** [review]
A few more comments (can be for the future)

**Comment by revit13** [comment]
@roee88  Thanks for the comments. I addressed them in the second commit.

**Review by roee88 — APPROVED** [review]

**Pull request merged**